### PR TITLE
Majo GetCard a BuildingTIle

### PR DESCRIPTION
**Pull request by nightmarra**
zmenil som state na JSON string, prislusne su upravene aj testy
vymazal som TODO komentare co nie su relevantne
v starom branchi by to bol chaos ked boli dalsie merge odvtedy, neviem ci by mi to islo tak som radsej urobil novy
prosim merge

### CODE REVIEW

File: src/test/java/sk/uniba/fmph/dcs/game_board/BuildingTileTest.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    @Override
+    public boolean addNewFigure() {
+        figureCount += 1;
+        return true;
+    }
+}
+
+public class BuildingTileTest {
+    @Test
+    public void testOccupiedState() {
```
Comment (alexdi906) [inline src/test/java/sk/uniba/fmph/dcs/game_board/BuildingTileTest.java:61]: tento test neprechadza

File: src/main/java/sk/uniba/fmph/dcs/game_board/BuildingTile.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    @Override
+    public HasAction tryToMakeAction(Player player) {
+        boolean playerFigureOnTile = false;
+        for(PlayerOrder figure: figures) {
+            if(figure.getOrder() == player.getPlayerOrder().getOrder()) {
+                playerFigureOnTile = true;
+                break;
+            }
+        }
+        if(!playerFigureOnTile || skipAction(player)) {
```
Comment (alexdi906) [inline src/main/java/sk/uniba/fmph/dcs/game_board/BuildingTile.java:93]: toto by myslim nemalo volat skipAction(), leby tryToMakeAction() sa vola este predtym ako hrac urobit akciu, takze aj keby tu hrac mohol urobit akciu, bude skipnuta

**Review by alexdi906 — COMMENTED** [review]
okrem tychto dvoch veci good

File: src/main/java/sk/uniba/fmph/dcs/game_board/BuildingTile.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    public HasAction tryToPlaceFigures(Player player, int count) {
+        if(figures.size() == 1 || (count != 0 && count != 1) || !player.getPlayerBoard().hasFigures(count)) {
+            return HasAction.NO_ACTION_POSSIBLE;
+        }
+        return HasAction.WAITING_FOR_PLAYER_ACTION;
+    }
+
+    @Override
+    public ActionResult makeAction(Player player, Collection<Effect> inputResources, Collection<Effect> outputResources) {
+        // TODO move scoring marker – na toto potrebujem class GameBoard s konkretnou verziou Stringu state
```
Comment (alexdi906) [inline src/main/java/sk/uniba/fmph/dcs/game_board/BuildingTile.java:63]: asi to staci urobit takto, (kopirujem ako sa hybe na agricultural track z triedy ToolMakerHutFields, teda
tu das asi Effect.Building)

player.getPlayerBoard().giveEffect(new Effect[] {Effect.FIELD});

**Review by alexdi906 — COMMENTED** [review]
este som si pri review ToolMakerHutFields vsimol toto

File: src/main/java/sk/uniba/fmph/dcs/game_board/BuildingTile.java (hunk 2 above)
Comment (nightmarra) [reply src/main/java/sk/uniba/fmph/dcs/game_board/BuildingTile.java:93]: v game rules mame ze "If a player cannot or does not want to pay the resources (co su na building tile), she takes back her figure and leaves the building in place." - je pre playera v nejakej classe implementovane ze sa vie rozhodnut takto skipnut?

File: src/test/java/sk/uniba/fmph/dcs/game_board/BuildingTileTest.java (hunk 1 above)
Comment (nightmarra) [reply src/test/java/sk/uniba/fmph/dcs/game_board/BuildingTileTest.java:61]: fixnem, nevedel som ho spustit predtym nez sa mergol ten pull request kde si fixol conflicting nazvy classov

File: src/main/java/sk/uniba/fmph/dcs/game_board/BuildingTile.java (hunk 3 above)
Comment (nightmarra) [reply src/main/java/sk/uniba/fmph/dcs/game_board/BuildingTile.java:63]: ok fixnem

File: src/main/java/sk/uniba/fmph/dcs/game_board/BuildingTile.java (hunk 2 above)
Comment (alexdi906) [reply src/main/java/sk/uniba/fmph/dcs/game_board/BuildingTile.java:93]: ano v  InterfaceGamePhaseController (co vlastne riadi hru) mozes urobit skipAction()

Comment (alexdi906) [reply src/main/java/sk/uniba/fmph/dcs/game_board/BuildingTile.java:93]: vlastne v tom InterfaceFigureLocationInternal vsetky metody co nezacinaju try a newTurn(), volaju hraci ked "hraju" hru.

**Pull request closed without merging**